extensions/amp-form/0.1/form-validators.js

```diff
@@ -468,10 +468,21 @@ export function isCheckValiditySupported(doc) {
  * @return {?string}
  */
 function getInvalidType(input) {
+  let isValueMissingError = false,
+      errorValue = null;
   for (const invalidType in input.validity) {
-    if (input.validity[invalidType]) {
+    if (invalidType === 'valueMissing' && input.validity[invalidType]) {
```

Comment:
@aghassemi Just to summarize the suggested approach. We would push all the `error types` in an array. Iterator over it and return the array item for which corresponding value in `input.validity` is `true`?

If yes, may I know this approach is more beneficial in terms of code understanding or is there any other benefit also (Just wanted to know)? Thanks!

Reply:
Hi @yomeshgupta, the benefit is mostly moving the logic from code (e.g. if/else in the middle of a fairly busy code) to the concept of "this is the ordered list of error types". helps with readability.

The code I had only gives precedence over `badInput` since it is the first one pushed in the array. I looked and could not find anything in the official spec that defines precedence of these errors. There might be more cases but given this solves the problem in hand and it is not clear if rearranging these any further may cause regression issues, I suggest to just do `badInput` for now.

Reply:
Hi @aghassemi Thank you for sharing the thought process behind the suggestions. I have made the necessary changes. Kindly, let me know if anything else is required from my end. Thanks!

Reply:
Looks good! Thank you